docs/gateway-api-compatibility.md

```diff
@@ -4,22 +4,21 @@ This document describes which Gateway API resources NGINX Kubernetes Gateway sup
 
 ## Summary
 
-| Resource                            | Support Status      |
-|-------------------------------------|---------------------|
-| [GatewayClass](#gatewayclass)       | Partially supported |
-| [Gateway](#gateway)                 | Partially supported |
-| [HTTPRoute](#httproute)             | Partially supported |
-| [TLSRoute](#tlsroute)               | Not supported       |
-| [TCPRoute](#tcproute)               | Not supported       |
-| [UDPRoute](#udproute)               | Not supported       |
-| [ReferenceGrant](#referencegrant)   | Supported           |
-| [Custom policies](#custom-policies) | Not supported       |
```

Comment:
Since at least for HTTPRoute, we already support Extended features, shall we update the table to something like below? (that is also what https://github.com/nginxinc/nginx-kubernetes-gateway/issues/605 covers)

---


| Resource                            |  Support level  - Core | Support level - Extended  | Support level - Implementation-specific  | API Version |
|-------------------------------------|---------------|-------------|--|--|
| [GatewayClass](#gatewayclass)       | Supportred | Partially supported | Not supported    | v1beta1     |
| [Gateway](#gateway)                 | Supported          | Not supported  | Not supported     | v1beta1 |

. . . other resources

and the Terminology section

## Terminology

Gateway API features has [three levels](https://gateway-api.sigs.k8s.io/concepts/conformance/#2-support-levels) of support: Core, Extended and Implementation-specific. We use the following words to describe support status per each level:
- *Supported*. The resource or field is fully supported and conformant to the Gateway API specification.
- *Partially supported*.  The resource or field is supported partially or with limitations.
- *Not supported*. The resource or field is not supported.
- *N/A*. The resource doesn't have the particular support level.

> We will add more features in the future releases. However, it might be possible that NGINX Kubernetes Gateway will never support some resources and/or fields of the Gateway
API. We will document these decisions on a case by case basis.

> NGINX Kubernetes Gateway doesn't support any features from the experimental [release channel](https://gateway-api.sigs.k8s.io/concepts/conformance/#1-release-channels).

---

Considering that suggestion, it also entails adding levels of supports to resources and fields.

### HTTPRoute

> Support Level - Core: Supported.
> Support Level - Extended: Partially supported.
> Support Level - Implementation: Not supported.

. . .

Perhaps the following can work to represent support level in fields?

Fields:

* `spec`
    * `parentRefs`  -- Core: supported; Extended (port): not supported.
    * `hostnames` - Core: supported.
    * `rules`
        * `matches`
            * `path` - Core ( `PathPrefix` and `Exact` ): supported. Implementation-specific (`RegularExpression` type): not supported.
            . . .
            * `method` -  Extended: supported.

---

Note: in the Gateway API community, there are some related issues:
- Conformance profiles. It is possible to report feature support based on the conformance profiles, which are based on the conformance tests. https://github.com/kubernetes-sigs/gateway-api/pull/1967#pullrequestreview-1408166850 and https://github.com/kubernetes-sigs/gateway-api/pull/2066  (*Question*: perhaps we can also find a way to visualize that data rather creating our own tables? )
- Currently in progress - reporting supported features in the GatewayClass -- https://github.com/kubernetes-sigs/gateway-api/pull/2163

Reply:
I like the suggestion for the table and terminology 👍 

However, I don't think we should specify the support level for the fields for a few reasons:
- Not every field has a corresponding support level 
- It makes it harder to read and understand. I think we want to keep the field status descriptions as simple as possible so people can quickly get the answer they need - which is simply "does NKG support X field?". 
- It makes it harder for us to maintain.

Reply:
> Not every field has a corresponding support level
> It makes it harder to read and understand. I think we want to keep the field status descriptions as simple as possible so people can quickly get the answer they need - which is simply "does NKG support X field?".

good points 

> It makes it harder for us to maintain.

but how do we know when to update the high level table if we don't know which fields of a support level we support and don't support in resource type?

Reply:
We just have to rely on the spec and cross-reference that with our current supported fields.

Reply:
The spec should be our source of truth. I don't want to re-create it here. I think that will be more work in the long-run.

Reply:
> We just have to rely on the spec and cross-reference that with our current supported fields.
> The spec should be our source of truth. I don't want to re-create it here. I think that will be more work in the long-run.

👍